### DISCUSSION TITLE
Migration from graphql-java spring-boot to DGS

### DISCUSSION THREAD

**Discussion by mattdsteele** [Netflix/dgs-framework#909] [Q&A] [answered]
Hi folks, are there any good resources to take an existing GraphQL server that was written up using the graphql-spring-boot project over to DGS? I've got a medium-sized app that I'm interesting in porting over to take advantage of the annotation-based model, federation, and other features.

Most of the Getting Started guides I see are focused on generating a brand new app using Initializr. Is this the recommended approach? Or are there paths to take existing `QueryResolvers` and moving them into DGS?

https://github.com/graphql-java-kickstart/graphql-spring-boot

**Comment by srinivasankavitha** [comment, marked as answer]
Currently, you will need to reimplement your code to use DGS annotations if you want to switch over. You shouldn't need to regenerate an entirely new project, but that depends on how your project is set up. Towards the end of this year, we will be looking to support using the DGS along with spring-graphql so you can choose between the features offered by both.
